stan/math/prim/fun/select.hpp

```diff
@@ -0,0 +1,140 @@
+#ifndef STAN_MATH_PRIM_FUN_SELECT_HPP
+#define STAN_MATH_PRIM_FUN_SELECT_HPP
+
+#include <stan/math/prim/meta.hpp>
+
+namespace stan {
+namespace math {
+
+/**
+ * Return the second argument if the first argument is true
+ * and otherwise return the third argument.
+ *
+ * <code>select(c, y1, y0) = c ? y1 : y0</code>.
+ *
+ * @tparam T_true type of the true argument
+ * @tparam T_false type of the false argument
+ * @param c Boolean condition value.
+ * @param y_true Value to return if condition is true.
+ * @param y_false Value to return if condition is false.
+ */
+template <typename T_true, typename T_false,
+          require_all_stan_scalar_t<T_true, T_false>* = nullptr>
+inline auto select(const bool c, const T_true y_true, const T_false y_false) {
+  return c ? y_true : y_false;
+}
```

Comment:
Not sure if @t4c1 still checks github, but I'm not sure if we need `common_type` here or if auto is fine? I wouldn't mind just using `return_type_t<>`, though that will only work with arithmetic types since `return_type_t` has a minimum of `double` as the returned type. We could just write another another overload to handle the double integral case though

Reply:
I still get notifications if pinged. `auto` will here be same as `T_true` (that is how ternary operator works), so some common type is a better idea. Not sure if `retrun_type` will do promotion to var even if neither T_true nor T_false are var, but we do not want that here.

Reply:
> `auto` will here be same as `T_true`

I've done some tests and it doesn't look like an issue when mixing types: https://godbolt.org/z/dvcxvvxhs

But let me know if I've missed something basic!